dimod/quadratic/quadratic_model.py

```diff
@@ -1224,5 +1224,41 @@ def Integers(labels: Union[int, Iterable[Variable]],
     else:
         yield from (Integer(dtype=dtype) for _ in range(labels))
 
+
+def IntegerArray(labels: Union[int, Iterable[Variable]],
+                 dtype: Optional[DTypeLike] = None) -> np.ndarray:
+    """Return a NumPy array of quadratic models, each with a 
+    single integer variable.
+
+    Args:
+        labels: Either an iterable of variable labels or a number. If a number
+            labels are generated using :class:`uuid.UUID`.
+        dtype: Data type for the returned quadratic models.
+
+    Returns:
+        Array of quadratic models, each with a single integer variable.
+    
+    """
+    return _VariableArray(Integers, labels, dtype)
+
+
+def _VariableArray(variable_generator: Callable,
+                   labels: Union[int, Iterable[Variable]],
+                   dtype: Optional[DTypeLike] = None) -> np.ndarray:
+    """Builds NumPy array from a variable generator method."""
+    if isinstance(labels, int):
+        number_of_elements = labels
+    elif isinstance(labels, Sized):
+        number_of_elements = len(labels)
+    else:
+        labels = list(labels)
```

Comment:
can we add a test for this case? Doesn't have to be for all of them, but at least one.

Reply:
Yeah, will add that.